docs/content/docs/tutorials/ec2-example.md

```diff
@@ -0,0 +1,450 @@
+natgateway.ec2.services.k8s.aws/tutorial-natgateway1 created
+elasticipaddress.ec2.services.k8s.aws/tutorial-eip1 created
+routetable.ec2.services.k8s.aws/tutorial-public-route-table created
+routetable.ec2.services.k8s.aws/tutorial-private-route-table-az1 created
+subnet.ec2.services.k8s.aws/tutorial-public-subnet1 created
+subnet.ec2.services.k8s.aws/tutorial-private-subnet1 created
+securitygroup.ec2.services.k8s.aws/tutorial-security-group created
+```
+ 
+* Check the **Custom Resource's** using `kubectl describe`:
+```
+kubectl describe vpcs
+kubectl describe internetgateways
+kubectl describe routetables
+kubectl describe natgateways
+kubectl describe elasticipaddresses
+kubectl describe subnets
+kubectl describe securitygroups
+```
+ 
+* Subnet gets into an 'available' state with a `ACK.ReferencesResolved = True` condition attached notifying users that the references (VPC, RouteTable) have been found and resolved:
+ 
+```
+Status:
+  Ack Resource Metadata:
+    Arn:                       arn:aws:ec2:us-west-2:<ID>:subnet/subnet-0ba22f5820bb41584
+    Owner Account ID:          <ID>
+    Region:                    us-west-2
+  Available IP Address Count:  4091
+  Conditions:
+    Last Transition Time:           2022-10-13T14:54:39Z
+    Status:                         True
+    Type:                           ACK.ReferencesResolved
+    Last Transition Time:           2022-10-13T14:54:41Z
+    Message:                        Resource synced successfully
+    Reason:
+    Status:                         True
+    Type:                           ACK.ResourceSynced
+  Default For AZ:                   false
+  Map Customer Owned IP On Launch:  false
+  Owner ID:                         515336597380
+  Private DNS Name Options On Launch:
+  State:      available
+  Subnet ID:  subnet-<ID>
+ 
+```
+ 
+### Validate
+ 
+This network setup should allow Instances deployed in the Private Subnet to connect to the internet. To validate this behavior deploy an Instance into the Private Subnet and the Public Subnet (bastion host). After deployments, `ssh` into the bastion host, then `ssh` into the Private Subnet Instance, and test internet connection. Security group is required by both instances launched in public and private subnets.
+ 
+* Deploy an Instance into the Private Subnet:
+ 
+```
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: Instance
+metadata:
+  name: tutorial-instance-private
+spec:
+  imageID: ami-02b92c281a4d3dc79 # AL2; us-west-2
+  instanceType: c3.large
+  subnetID: subnet-<private-ID>
+  securityGroupIDs:
+  	- sg-<ID>
+  keyName: us-west-2-key # created via console
+  tags:
+    - key: producer
+      value: ack
+```
+ 
+* Deploy an Instance into the Public Subnet:
+ 
+```
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: Instance
+metadata:
+  name: tutorial-instance-public
+spec:
+  imageID: ami-02b92c281a4d3dc79 # AL2 in us-west-2
+  instanceType: c3.large
+  subnetID: subnet-<public-ID>
+  securityGroupIDs:
+  	- sg-<ID>
```

Comment:
there is a formatting error here
`dsaurya@3c063003223c ~ % k apply -f tutorial-instance-public.yaml
error: error parsing tutorial-instance-public.yaml: error converting YAML to JSON: yaml: line 10: found character that cannot start any token`

Reply:
cross-checked the yaml file, there is an indentation error while adding `SecurityGroupIDs`, will fix this!